sycl/source/detail/queue_impl.cpp

```diff
@@ -801,6 +802,7 @@ void queue_impl::revisitUnenqueuedCommandsState(
   else {
     std::lock_guard<std::mutex> RequestLock(MMissedCleanupRequestsMtx);
     MMissedCleanupRequests.push_back(CompletedHostTask->getCommandGraph());
+    MAreCleanupRequestsMissed.store(true, std::memory_order_release);
```

Comment:
Why is the following not an issue:
```
Thread A        Thread B
acquire mtx
                check atomic without mutex, empty
push_back
atomic_store_true
```
?

Reply:
Thank you, this is exactly the question I would like to discuss with the fellow sycl experts! 

According to my understanding, the change is correct, because the situation you describing is undistinguishing from a situation when Thread A just staying, say, at the beginning of `queue_impl::revisitUnenqueuedCommandsState()`, i.e. there is no reason for Thread B to expect something in `MMissedCleanupRequests` at this point.

Reply:
@sergey-semenov , @KseniyaTikhomirova would be much more knowledgeable about this than I am.

Reply:
Chatted with @KseniyaTikhomirova about this today. Like Alexandr said, the case is indistinguishable from the first thread being earlier in its execution path. 
With the synchronization being made less strict, it will be slightly more likely that we'll exit `queue::wait` with some dependency related data still present in the queue. But that's not a functional problem since that data will be handled during the next cleanup call.